### PR TITLE
[DCOS-1192] Catch exceptions when parsing config file

### PR DESCRIPTION
**Pull request by mgummelt**

### CODE REVIEW

File: dcos/config.py [hunk 1]
```diff
@@ -14,7 +15,11 @@ def mutable_load_from_path(path):
     """
 
     with util.open_file(path) as config_file:
-        return MutableToml(toml.loads(config_file.read()))
+        try:
+            return MutableToml(toml.loads(config_file.read()))
```
Comment (tamarrow) [inline dcos/config.py]: Were we going to add the file if it doesn't exist?

Comment (mgummelt) [inline dcos/config.py]: that's in a different PR

**Comment by mgummelt** [comment]
moved @jsancio to reviewer because he is the author

File: dcos/config.py [hunk 2]
```diff
@@ -14,7 +15,11 @@ def mutable_load_from_path(path):
...
+        except Exception as e:
```
Comment (jsancio) [inline dcos/config.py]: Please add `logger.exception(...)`

File: dcos/config.py [hunk 3]
```diff
@@ -27,7 +32,11 @@ def load_from_path(path):
     """
 
     with util.open_file(path) as config_file:
-        return Toml(toml.loads(config_file.read()))
+        try:
+            return Toml(toml.loads(config_file.read()))
+        except Exception as e:
```
Comment (jsancio) [inline dcos/config.py]: Please add `logger.exception(...)`

**Comment by jsancio** [comment]
Thanks. Looks good. Can we add tests for this PR?

File: dcos/config.py (hunk 2 above)
Comment (mgummelt) [inline dcos/config.py]: When in general should we use `logger.exception`?  Can you add that to the review wiki?

I understand adding it if there's valuable info that isn't contained in the exception message, but that's not the case here.

**Comment by mgummelt** [comment]
added test

**Pull request merged**